compiler/rustc_target/src/abi/mod.rs

```diff
@@ -610,6 +616,17 @@ impl Integer {
         }
         I8
     }
+
+    fn from_bits(bits: u64) -> Result<Self, String> {
+        match bits {
+            8 => Ok(Integer::I8),
+            16 => Ok(Integer::I16),
+            32 => Ok(Integer::I32),
+            64 => Ok(Integer::I64),
+            128 => Ok(Integer::I128),
+            _ => Err(format!("rust does not support integers with {} bits", bits)),
+        }
+    }
```

Comment:
This is probably the addition I care most about. I wonder if there's a suitable replacement around? I remember avoiding having to add such a function over time.

It's likely fine, I just haven't had time to think about it.

Maybe at least have it take `Size`? That way the "bits" naming is handled separately.